analyzers/src/SonarAnalyzer.Common/Helpers/UnitTestHelper.cs

```diff
@@ -18,111 +18,119 @@
+    /// </summary>
+    public static bool IsTestClass(this INamedTypeSymbol classSymbol) =>
+        classSymbol.AnyAttributeDerivesFromAny(KnownTestClassAttributes);
+
+    public static bool IsTestMethod(this IMethodSymbol method) =>
+        method.MethodKind.HasFlag(MethodKindEx.LocalFunction)
+            ? method.IsXunitTestMethod()
+            : method.AnyAttributeDerivesFromOrImplementsAny(KnownTestMethodAttributes);
+
+    public static bool IsIgnoredTestMethod(this IMethodSymbol method) =>
+        method.IsMsTestOrNUnitTestIgnored()
+        || method.FindXUnitTestAttribute().NamedArguments.Any(arg => arg.Key == "Skip");
+
+    public static bool HasExpectedExceptionAttribute(this IMethodSymbol method) =>
+        method.GetAttributes().Any(a =>
+            a.AttributeClass.IsAny(KnownExpectedExceptionAttributes)
+            || a.AttributeClass.DerivesFrom(KnownType.Microsoft_VisualStudio_TestTools_UnitTesting_ExpectedExceptionBaseAttribute));
+
+    public static bool HasAssertionInAttribute(this IMethodSymbol method) =>
+        !NoExpectedResultTestMethodReturnTypes.Any(method.ReturnType.Is)
+        && method.GetAttributes().Any(IsAnyTestCaseAttributeWithExpectedResult);
+
+    public static AttributeData FindXUnitTestAttribute(this IMethodSymbol method) =>
+        method.GetAttributes().FirstOrDefault(a =>
+            a.AttributeClass.Is(KnownType.Xunit_FactAttribute)
+            || a.AttributeClass.Is(KnownType.Xunit_TheoryAttribute)
+            || a.AttributeClass.Is(KnownType.LegacyXunit_TheoryAttribute));
+
+    /// <summary>
+    /// Returns the <see cref="KnownType"/> that indicates the type of the test method or
+    /// null if the method is not decorated with a known type.
+    /// </summary>
+    /// <remarks>We assume that a test is only marked with a single test attribute e.g.
+    /// not both [Fact] and [Theory]. If there are multiple attributes only one will be
+    /// returned.</remarks>
+    public static KnownType FindFirstTestMethodType(this IMethodSymbol method) =>
+        KnownTestMethodAttributes.FirstOrDefault(known =>
+                method.GetAttributes().Any(att => att.AttributeClass.DerivesFrom(known)));
+
+    private static bool IsAnyTestCaseAttributeWithExpectedResult(AttributeData a) =>
+        IsTestAttributeWithExpectedResult(a)
+        || a.AttributeClass.Is(KnownType.NUnit_Framework_TestCaseSourceAttribute);
+
+    private static bool IsTestAttributeWithExpectedResult(AttributeData a) =>
+        a.AttributeClass.IsAny(KnownType.NUnit_Framework_TestCaseAttribute, KnownType.NUnit_Framework_TestAttribute)
+        && a.NamedArguments.Any(arg => arg.Key == "ExpectedResult");
+
+    private static bool IsXunitTestMethod(this IMethodSymbol methodSymbol) =>
+       methodSymbol.AnyAttributeDerivesFromAny(KnownTestMethodAttributesOfxUnit);
+
+    private static bool IsMsTestOrNUnitTestIgnored(this IMethodSymbol method) =>
```

Comment:
To match the logic of the field that it depends on
```suggestion
    private static bool HasIgnoredAttribute(this IMethodSymbol method) =>
```